GOVERNANCE.md

```diff
@@ -0,0 +1,118 @@
+- The health of the project going forward
+
+Overall, these 'project leads' should not be beholden to their respective companies, and instead to the health of the project and its community.
+
+Underneath the leads are the teams or individuals, that each are allowed to  make similar decisions about the project *in the area they have been delegated*.
+
+
+- Some decisions are global in the project, like the core/API
+    - Team members should try and make sure to explore possible options together before calling for a vote. This mostly includes uncontroversial changes that are low impact or have already been agreed upon
+    - If this is not possible, a vote needs to be held: TODO
+
+- Some decisions are local, e.g. a plugin does not impact others with its behaviour, however it still needs to be a 'good citizen' in the project (CI, format, etc..)
+    - These should be clearly scoped in their respective project part
+    - However, if needed a 'higher up' decision can be requested if no consensus is achieved
+
+
+
+## Project Structure
+
+The project has a hierarchy to solve these two problems:
+
+1. How can participating organizations make sure that their voice and interests are represented in the project?
+    - Each member party appoints one core team member
+    - They form the group of people that have to, and are responsible for, signing off on the technical implementation of the project
+2. How to make the lives of these team members easier?
+
+
+-----
+
+* Thin-edge is organized around sub-projects - to ease decision taking.
+    * The sub-projects are organized in a hierarchy
+        * This makes it easier to decouple different parts of the project
+        * This makes it easier to decide on technical questions
+    * Everyone has to agree on the core project bringing the foundations to all others.
+    * At the periphery, sub-projects might be related to specific eco-systems (e.g. Cumulocity or Moneo) and therefore have independent decision processes.
+    *	At an intermediate level, one might have sub-projects related to specific use-cases like “JSON over MQTT”.
+* All the sub-projects share a common repository - to ease consistency across projects and over time.
+    *	Labels are used to organize issues/PRs/discussions along sub-projects.
+    *	Code ownership is used to enforce cooperation around key components, but the default is to let things open and to trust each other, using version control as a safety net.
+
+
+
+
+
+
+```
+core contributors = [SAG, IFM]
+
+-> Project leads, one each from [core contributors]
+|
+ \- Responsible for `/*`, aka everything in the repository
+ |
+ |
+ \
+  \-> Delegated project Teams, any combination of [teams, individual contributors]
+  |
+  |\- Responsible for `/plugins/plugin_{foo,bar}`
+  ||- Are trusted by the project leads (who would have to stand-in for the trusted members)
+  | 
+  |->  Can be needed to nest further if the project grows bigger
+```
+
+## Repository maintenance
```

Comment:
Hey @didier-wenzek - would it be okay if I submit patches for this section, to fill the individual subsections? I would propose my ideas via a PR to your repository, so you can merge my changes onto this PR.

Reply:
Sure. We need to be specific on many points. Just beware we are still waiting for an agreement on the overall proposal from the different stakeholders.

Reply:
Yes, sure. I like to believe that the repository maintenance part does not have too much points where it might collide with things other stakeholders might suggest. :wink: